### PR TITLE
Fixed banding in the volumetric clouds (case 1353672).

### PR DESCRIPTION
**Pull request by anisunity**
https://fogbugz.unity3d.com/f/cases/1353672/
This PR fixes 3 type of issues.
- Artifacts when the density at height 0 or 1 is not null (when defined by the curve)
- Banding artifacts
- Cloud discontinuity

Testing status
Tried to test extensively, cannot reproduce any of the issues anymore.

### CODE REVIEW

**Review by pierre-unity — APPROVED** [review]
All good on my side.
- all 4 test scenes are working fine
- shadows have been fixed

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/VolumetricLighting/VolumetricClouds.compute [hunk 1]
```diff
@@ -366,25 +366,25 @@ struct CloudCoverageData
     float maxCloudHeight;
 };
 
-float3 AnimateCloudMapPosition(float3 positionWS)
+float3 AnimateCloudMapPosition(float3 positionPS)
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/VolumetricLighting/VolumetricClouds.compute:369]: what mean PS here?

Comment (anisunity) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/VolumetricLighting/VolumetricClouds.compute:369]: PlanetSpace

**Comment by anisunity** [comment]
Fixed formatting, adding tests

**Comment by anisunity** [comment]
need to update the screenshots

**Comment by anisunity** [comment]
Updated the screenshots and relaunched the tests

**Review by remi-chapelain — COMMENTED** [review]
It looks like the fix works properly, 
Here's two example from the tests scene that shows the issues. 

https://user-images.githubusercontent.com/57442369/127474090-afceade0-c229-4b52-859a-ee5043883b9c.mp4

https://user-images.githubusercontent.com/57442369/127474093-2a831b4c-5edd-48ef-8471-759e1885a995.mp4

As for the test, I've updated it, lessen the thresholds, put it more contrast and more light to be able to make it fail when the fix is not there (was still passing without the fix before). 
I've also removed wind in the tests for better robustness. 

Yamato has been relaunch, and I'll update the screenhots when it's done.

**Review by sebastienlagarde — APPROVED** [review]

**Review by remi-chapelain — APPROVED** [review]
Tests are not fully green (linux and osx still fail) but volumetric clouds pass, so we should be fine.

**Comment by anisunity** [comment]
i'll_rerun_test_untill_they_pass_exe

**Comment by anisunity** [comment]
All green

**Pull request merged**